src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/GoalController.kt

```diff
@@ -39,7 +41,9 @@ class GoalController(
     goalService.createGoal(
       prisonNumber = prisonNumber,
       goal = goalResourceMapper.fromModelToDomain(request),
-    )
+    ).apply {
+      telemetryService.trackGoalCreateEvent(this)
+    }
```

Comment:
This approach of injecting the `TelemetryService` into the controller will do for now, to prove it works and serve as an example, but I think we can do better moving forwards (this approach will get messy moving forwards)

I did consider injecting the `TelemetryService` into the `GoalService`, but the `GoalService` is a domain class/concern, and I really don't think we want the domain being dependent on things in the app (like this `TelemetryService` bean)
I did consider creating the `TelemetryService` in the domain, but that felt wrong, as sending telemetry events etc does not feel like a concern of the domain.
Hence I added it to the controller here.

I think moving forwards we can perhaps look at an AOP approach, where the `TelemetryService` is not injected into either the controller or the domain. That feels like a better separation perhaps, but we can look at that later

Reply:
I think AOP would be ideal for this 👍